### DISCUSSION TITLE
Clarification on the overall GSoC process

### DISCUSSION THREAD

**Discussion by krancour** [cncf/mentoring#621] [Google Summer of Code 2022]
I've been looking at the GSoC timeline [here](https://developers.google.com/open-source/gsoc/timeline) and it's created a small bit of confusion for me that I hope someone can clarify. Asking here in the open in case I am not the only one with questions!

It appears that from March 7 - April 3, "Potential GSoC contributors discuss application ideas with mentoring organizations."

This was surprising to me because I didn't see anything prior to that in the timeline where the project ideas that we proposed as a mentoring organization (CNCF collectively generated dozens of project ideas) were either accepted or rejected.

How then do potential contributors know which projects are truly available to apply for?

_Or..._

Is it more the case that _all_ project ideas we proposed are fair game and project idea + contributor pairs are selected as a _unit_ on the basis of their proposal?

I'd be so grateful for anyone who can clarify this for myself and others who are new to GSoC.

**Comment by nate-double-u** [comment]
Hi Kent, thanks for this!

> It appears that from March 7 - April 3, "Potential GSoC contributors discuss application ideas with mentoring organizations."
> This was surprising to me because I didn't see anything prior to that in the timeline where the project ideas that we proposed as a mentoring organization (CNCF collectively generated dozens of project ideas) were either accepted or rejected.

Google Summer of Code accepted the CNCF as a participating organization on March 7, and the [Google Summer of Code 2022 & CNCF](https://github.com/cncf/mentoring/discussions/565) thread is probably the closest thing we made to an announcement.

> How then do potential contributors know which projects are truly available to apply for?

I think that a potential contributor's proposal for a project _is_ their application.

> Is it more the case that all project ideas we proposed are fair game and project idea + contributor pairs are selected as a unit on the basis of their proposal?

We will need to review the GSoC contributor proposals; I'm not sure how we will do this yet. Once we review, the CNCF will start requesting contributor slots and assigning mentors to projects. So, once we get to the slots selection stage, we will have a pretty good idea of who would be working on and mentoring a project. Note, there isn't a guarantee that GSoC will grant all the slots requested, so we'll also need to decide the priority of any projects as we review.

Some resources that may help mentors make sense of [the timeline](https://developers.google.com/open-source/gsoc/timeline):
* [GSoC Organization Admin Tips](https://developers.google.com/open-source/gsoc/help/oa-tips)
* [Frequently Asked Questions](https://developers.google.com/open-source/gsoc/faq)
* [Guides](https://developers.google.com/open-source/gsoc/resources/guide#student_manual) (for both contributors/students, and mentors)

**Comment by krancour** [comment]
Another question I've heard from several people is whether applicants are permitted to have their would-be mentors review their applications prior to them being submitted.

**Reply by nate-double-u** [reply]
I think the [GSoC 2022 FAQ](https://developers.google.com/open-source/gsoc/faq) covers that with this question:

[What should I do to improve my chances of being selected as a GSoC contributor?](https://developers.google.com/open-source/gsoc/faq#what_should_i_do_to_improve_my_chances_of_being_selected_as_a_gsoc_contributor)

There is a time allotted for contributors to discuss projects with mentors and the community, and they should use those discussions to write up a proposal. Once a proposal is submitted there is time for mentors to give feedback. Contributors should submit their proposals as early as they can to ensure that they've got time to incorporate that feedback and resubmit before the deadline.
